src/sentry/migrations/0881_artifactbundleindex_org_bundle_url.py

```diff
@@ -0,0 +1,44 @@
+# Generated by Django 5.1.7 on 2025-05-02 21:48
+
+import django.contrib.postgres.indexes
+import django.db.models.functions.text
+from django.db import migrations, models
+
+from sentry.new_migrations.migrations import CheckedMigration
+
+
+class Migration(CheckedMigration):
+    # This flag is used to mark that a migration shouldn't be automatically run in production.
+    # This should only be used for operations where it's safe to run the migration after your
+    # code has deployed. So this should not be used for most operations that alter the schema
+    # of a table.
+    # Here are some things that make sense to mark as post deployment:
+    # - Large data migrations. Typically we want these to be run manually so that they can be
+    #   monitored and not block the deploy for a long period of time while they run.
+    # - Adding indexes to large tables. Since this can take a long time, we'd generally prefer to
+    #   run this outside deployments so that we don't block them. Note that while adding an index
+    #   is a schema change, it's completely safe to run the operation after the code has deployed.
+    # Once deployed, run these manually via: https://develop.sentry.dev/database-migrations/#migration-deployment
+
+    is_post_deployment = True
+
+    dependencies = [
+        ("sentry", "0880_orgauthtoken_bigint"),
+    ]
+
+    operations = [
+        migrations.AddIndex(
+            model_name="artifactbundleindex",
+            index=models.Index(
+                models.F("organization_id"),
+                models.F("artifact_bundle_id"),
+                django.contrib.postgres.indexes.OpClass(
+                    django.db.models.functions.text.Upper(
+                        django.db.models.functions.text.Reverse(models.F("url"))
+                    ),
+                    name="text_pattern_ops",
+                ),
+                name="sentry_artifactbundleindex_url",
+            ),
+        ),
+    ]
```

Comment:
We're creating this index manually via ops, since this table is huge in US. I'll merge this after the index is created.

Even without this index, the query as is will work fine, just at the same performance as before.